server/src/main/java/org/candlepin/model/EntitlementCurator.java

```diff
@@ -364,6 +481,33 @@ public Entitlement replicate(Entitlement ent) {
         return criteria.list();
     }
 
+    /**
+     * Batch deletes a list of entitlements.
+     * @param pools
+     */
+    public void batchDelete(List<Entitlement> entitlements) {
+        for (Entitlement ent : entitlements) {
+            log.debug("Deleting entitlement: " + ent);
+            log.debug("certs.size = " + ent.getCertificates().size());
+
+            for (EntitlementCertificate cert : ent.getCertificates()) {
+                getEntityManager().remove(cert);
+            }
+            ent.getCertificates().clear();
+            getEntityManager().remove(ent);
+        }
+
+        // Maintain runtime consistency.
+        for (Entitlement ent : entitlements) {
+            ent.getCertificates().clear();
+            // Pool.entitlements is EXTRA Lazy. Before removing anything
+            // from that collection, its necessary to initialize it
+            Hibernate.initialize(ent.getPool().getEntitlements());
```

Comment:
Just for my information, is this something that is commonly done? Is it just the nature of how our models are set up?